### ISSUE TITLE
"Open Rendered Image" greyed out & inactive Graphvizer panel

### ISSUE THREAD

**Issue by PaulPoinet** [hao-lee/Graphvizer#4]
Hello,
Thank you for developing this plug-in which i believe can be super helpful!
I've been going through all the steps you wrote to install and run it. Unfortunately, even though I installed it, it seems that its main functionalities do not work properly for me.
The "Open Rendered Image" command is greyed out (so I cannot run it), and the Graphvizer panel does not react to the code I write above. Any idea what I should do to fix it?
I'm running on windows 10.
Best
Paul

![bugs](https://user-images.githubusercontent.com/22736626/42545139-4bb0ac1e-84b6-11e8-93dc-052c2eaa0024.PNG)

**Comment by hao-lee** [comment]
Hi, this is indeed weird. This menuitem shouldn't grey out. Try `Ctrl+~` to open the Sublime Console and see if there are some errors about Graphvizer.

**Comment by PaulPoinet** [comment]
This is what i get:

DPI mode: per-monitor v2
startup, version: 3176 windows x64 channel: stable
executable: /C/Program Files/Sublime Text 3/sublime_text.exe
working dir: /C/Program Files/Sublime Text 3
packages path: /C/Users/Paul/AppData/Roaming/Sublime Text 3/Packages
state path: /C/Users/Paul/AppData/Roaming/Sublime Text 3/Local
zip path: /C/Program Files/Sublime Text 3/Packages
zip path: /C/Users/Paul/AppData/Roaming/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.0933511
startup time: 0.140351
first paint time: 0.140351
reloading plugin Default.arithmetic
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.colors
reloading plugin Default.comment
reloading plugin Default.convert_color_scheme
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.rename
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin ShellScript.ShellScript
reloading plugin Graphvizer.graphvizer
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 116, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 584, in _check_name_wrapper
  File "<frozen importlib._bootstrap>", line 1022, in load_module
  File "<frozen importlib._bootstrap>", line 1003, in load_module
  File "<frozen importlib._bootstrap>", line 560, in module_for_loader_wrapper
  File "<frozen importlib._bootstrap>", line 868, in _load_module
  File "<frozen importlib._bootstrap>", line 313, in _call_with_frames_removed
  File "C:\Users\Paul\AppData\Roaming\Sublime Text 3\Packages\Graphvizer\graphvizer.py", line 5, in <module>
    from . import syntaxchecker
  File "C:\Users\Paul\AppData\Roaming\Sublime Text 3\Packages\Graphvizer\syntaxchecker.py", line 1, in <module>
    from .antlr4 import *
  File "C:\Users\Paul\AppData\Roaming\Sublime Text 3\Packages\Graphvizer\antlr4\__init__.py", line 5, in <module>
    from .CommonTokenStream import CommonTokenStream
  File "C:\Users\Paul\AppData\Roaming\Sublime Text 3\Packages\Graphvizer\antlr4\CommonTokenStream.py", line 33, in <module>
    from .Lexer import Lexer
  File "C:\Users\Paul\AppData\Roaming\Sublime Text 3\Packages\Graphvizer\antlr4\Lexer.py", line 12, in <module>
    from typing.io import TextIO
ImportError: No module named 'typing'
reloading plugin Graphvizer.graphvizer_panel
reloading plugin Graphvizer.syntaxchecker
Traceback (most recent call last):
  File "C:\Program Files\Sublime Text 3\sublime_plugin.py", line 116, in reload_plugin
    m = importlib.import_module(modulename)
  File "./python3.3/importlib/__init__.py", line 90, in import_module
  File "<frozen importlib._bootstrap>", line 1584, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1565, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1532, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 584, in _check_name_wrapper
  File "<frozen importlib._bootstrap>", line 1022, in load_module
  File "<frozen importlib._bootstrap>", line 1003, in load_module
  File "<frozen importlib._bootstrap>", line 560, in module_for_loader_wrapper
  File "<frozen importlib._bootstrap>", line 868, in _load_module
  File "<frozen importlib._bootstrap>", line 313, in _call_with_frames_removed
  File "C:\Users\Paul\AppData\Roaming\Sublime Text 3\Packages\Graphvizer\syntaxchecker.py", line 1, in <module>
    from .antlr4 import *
  File "C:\Users\Paul\AppData\Roaming\Sublime Text 3\Packages\Graphvizer\antlr4\__init__.py", line 5, in <module>
    from .CommonTokenStream import CommonTokenStream
  File "C:\Users\Paul\AppData\Roaming\Sublime Text 3\Packages\Graphvizer\antlr4\CommonTokenStream.py", line 33, in <module>
    from .Lexer import Lexer
  File "C:\Users\Paul\AppData\Roaming\Sublime Text 3\Packages\Graphvizer\antlr4\Lexer.py", line 12, in <module>
    from typing.io import TextIO
ImportError: No module named 'typing'
plugins loaded

**Comment by PaulPoinet** [comment]
"No module named typing", does it come from there?

**Comment by hao-lee** [comment]
Yes, which Sublime version do you use?

**Comment by PaulPoinet** [comment]
3

**Comment by PaulPoinet** [comment]
does it have something to do with the version of python that is installed on my laptop? Originally I had 3.5 and installed 3.7, but it seems that the typing package is included in both of them, from what i've read online...

**Comment by hao-lee** [comment]
Sublime uses a Python that is custom for itself and this Python package is contained in Sublime directory, so you system-wide Python won't be used by Sublime.
Sublime before Build 3158 doesn't have `enum` and `typing` module and I have suggested the author to add them when I submitted this plugin.
Update Sublime to the latest version should fix this problem.

**Comment by PaulPoinet** [comment]
Hm.. Weird, I downloaded sublime today from the official website (Sublime Text 3 (Build 3176)), so > 3158. But indeed I looked at the installation file, checked the python dll and zip folder, I cannot find any typing module there...

**Comment by PaulPoinet** [comment]
the python version shipped with the latest version of sublime is 3.3

**Comment by hao-lee** [comment]
`Ctrl+Shift+P` then run `Satisfy Dependencies` and restart Sublime. :joy:

**Comment by PaulPoinet** [comment]
Lol this was a joke? How come am I the first one reporting such issue? Isn't your python version shipped with sublime also 3.3 ?

**Comment by PaulPoinet** [comment]
Any hint on how I can upgrade the python version within sublime.. ?

**Comment by hao-lee** [comment]
> Lol this was a joke? How come am I the first one reporting such issue? Isn't your python version shipped with sublime also 3.3 ?

Yes. The Python that shipped with Sublime is 3.3 and it won't be update for a long time in the future. The author will only add some modules into it but won't update it to Python-3.6. You are the first one report this issue.
When install this plugin, Sublime should install all dependencies automatically. However, I don't know why your Sublime doesn't do that correctly. :smirk:

**Comment by hao-lee** [comment]
> Any hint on how I can upgrade the python version within sublime.. ?

There should be some ways to update the Python embedded in Sublime, but the author says that it would break many plugins.

**Comment by PaulPoinet** [comment]
Well, I think it was my fault from start... Initially I was trying to install it manually because I wasn't familiar with the package control command. I finally went with it and it solved everything! Now it works super smooth and it's quite a pleasure to see the graph adapt in real time to code modifications 😄. Thanks for your help and for this plugin!

**Comment by hao-lee** [comment]
You are welcome. Maybe I should add some instructions to the `Manually` section in README.md to help others install dependencies. I didn't consider this problem before. :joy:

**Comment by PaulPoinet** [comment]
Yeah... I mean it seems that for now the manual install that you are describing is a simple drag n drop procedure and I don't see how this is affecting python.

In any case I would recommend to delete the "You must have known how to do this." mention from the Package Control install instructions, because when I saw this I actually skipped it and went for the manual instructions. But it is actually super straightforward and the indications you wrote below this mention are more than enough! (it was the first time i was installing a package in sublime so you are dealing with a real noob here 🤣)
